src/main/java/org/codehaus/mojo/build/CreateMojo.java

```diff
@@ -711,7 +711,7 @@ protected String filterBranchFromScmUrl( String scmUrl )
     {
         String scmBranch = "UNKNOWN";
 
-        if ( StringUtils.contains( scmUrl, "/trunk" ) )
+        if ( StringUtils.contains( scmUrl, "/trunk" ) && scmUrl.matches( ".*/trunk(/.*)?$" ) )
```

Comment:
What do you think about adding an option like boolean to activate this behaviour. This would keep compatibility and also gives other to change this if needed.

Reply:
Sure, but I cannot think of a case where the user would expect a branch named "trunkfoo" to be reported as "trunk". 

What might this option be called? Something like "allow branch names starting with trunk"?